inttest/openebs/openebs_test.go

```diff
@@ -0,0 +1,197 @@
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type OpenEBSSuite struct {
+	common.BootlooseSuite
+}
+
+func (s *OpenEBSSuite) TestK0sGetsUp() {
+	ctx := s.Context()
+	s.T().Run("Both storage and helm extensions enabled", func(t *testing.T) {
+		s.PutFile(s.ControllerNode(0), "/tmp/k0s.yaml", k0sConfigWithBoth)
+		s.Require().NoError(s.InitController(0, "--config=/tmp/k0s.yaml", "--disable-components=konnectivity-server,metrics-server"))
+		s.Require().NoError(s.RunWorkers())
+
+		kc, err := s.KubeClient(s.ControllerNode(0), "")
+		s.Require().NoError(err)
+
+		s.Require().NoError(s.WaitForNodeReady(s.WorkerNode(0), kc))
+
+		// When both storage and helm are enabled, there should be no action.
+		// Unfortunately to check that, the best we can do is seeing that it's not
+		// created after a grace period
+		s.T().Log("Waiting 30 additional seconds of grace period to see if charts are created")
+		ctx := s.Context()
+		select {
+		case <-ctx.Done():
+			s.Require().NoError(ctx.Err())
+		case <-time.After(30 * time.Second):
+		}
+
+		s.T().Log("Checking that the chart isn't created")
+		hc, err := s.HelmClient(s.ControllerNode(0), "")
+		s.Require().NoError(err)
+		_, err = hc.Charts("kube-system").Get(ctx, openEBSChart, metav1.GetOptions{})
+		s.Require().True(errors.IsNotFound(err), "Chart was created when it shouldn't have been")
+
+		s.T().Log("Stopping k0s")
+		s.Require().NoError(s.StopController(s.ControllerNode(0)))
+	})
+
+	s.T().Run("Only storage extension enabled", func(t *testing.T) {
+		s.PutFile(s.ControllerNode(0), "/tmp/k0s.yaml", k0sConfigWithStorage)
+		s.Require().NoError(s.StartController(s.ControllerNode(0)))
+
+		kc, err := s.KubeClient(s.ControllerNode(0), "")
+		s.Require().NoError(err)
+
+		s.Require().NoError(s.WaitForNodeReady(s.WorkerNode(0), kc))
+
+		s.T().Log("Checking that the chart is created and ready")
+		s.Require().NoError(s.waitForChartUpdated(ctx, "3.3.0"))
+		s.waitForOpenEBSReady(ctx, kc)
+
+		s.T().Log("Stopping k0s")
+		s.Require().NoError(s.StopController(s.ControllerNode(0)))
+	})
+
+	s.T().Run("Remove extension and upgrade", func(t *testing.T) {
+		s.PutFile(s.ControllerNode(0), "/tmp/k0s.yaml", k0sConfigNoExtension)
+		s.Require().NoError(s.StartController(s.ControllerNode(0)))
+
+		kc, err := s.KubeClient(s.ControllerNode(0), "")
+		s.Require().NoError(err)
+
+		s.Require().NoError(s.WaitForNodeReady(s.WorkerNode(0), kc))
+
+		s.T().Log("Updating OpenEBS chart to 3.9.0")
+		hc, err := s.HelmClient(s.ControllerNode(0), "")
+		s.Require().NoError(err)
+
+		c, err := hc.Charts("kube-system").Get(ctx, openEBSChart, metav1.GetOptions{})
+		s.Require().NoError(err, "Error getting OpenEBS chart after removing the storage extension")
+		c.Spec.Version = "3.9.0"
+
+		_, err = hc.Charts("kube-system").Update(ctx, c, metav1.UpdateOptions{})
+		s.Require().NoError(err, "Error updating OpenEBS chart")
```

Comment:
I think we need to test more than that, as mentioned [here](https://github.com/k0sproject/k0s/pull/3651/files#r1424036710).

How will the managed extension file will disappear from the helm folder of existing controllers?
How it will be prevented to be purged by the manifest applier?

If we leave this as manual migration steps for users, we need to augment the migration guide and mimic the described steps in this inttest to ensure that they work as expected.

Reply:
> How will the managed extension file disappear from the helm folder of existing controllers?

Ah I get it now, I'll add a restart and check that the version didn't roll back.

Reply:
I added the steps to remove the annotation to the docs and remove the manifest as well, even though the steps aren't really required I guess it could provoke some accidental damage.

Also I added it to the test